iree/compiler/Dialect/Flow/Transforms/test/expand_tensor_shapes.mlir

```diff
@@ -0,0 +1,155 @@
+  util.do_not_optimize(%0#0) : tensor<4x?x?x2xf32>
+  // CHECK-NEXT: util.do_not_optimize(%[[TIED_RET1]])
+  util.do_not_optimize(%0#1) : tensor<?xi32>
+
+  return
+}
+
+func private @callee(%arg0: tensor<4x?x?x2xf32>, %arg1: tensor<?xi32>) -> (tensor<4x?x?x2xf32>, tensor<?xi32>)
+
+// -----
+
+// Tests that branch arguments are expanded.
+
+// CHECK-LABEL: @br
+// CHECK-SAME: (%[[ARG0:.+]]: tensor<4x?x?x2xf32>, %[[ARG0_D1:.+]]: index, %[[ARG0_D2:.+]]: index,
+// CHECK-SAME:  %[[ARG1:.+]]: tensor<?xi32>, %[[ARG1_D0:.+]]: index)
+func @br(%arg0: tensor<4x?x?x2xf32>, %arg1: tensor<?xi32>) {
+  // CHECK-NEXT: %[[TIED_ARG0:.+]] = flow.tensor.tie_shape %[[ARG0]] : tensor<4x?x?x2xf32>{%[[ARG0_D1]], %[[ARG0_D2]]}
+  // CHECK-NEXT: %[[TIED_ARG1:.+]] = flow.tensor.tie_shape %[[ARG1]] : tensor<?xi32>{%[[ARG1_D0]]}
+
+  // CHECK-NEXT: cf.br ^bb1(%[[ARG0]], %[[ARG0_D1]], %[[ARG0_D2]], %[[ARG1]], %[[ARG1_D0]]
+  cf.br ^bb1(%arg0, %arg1 : tensor<4x?x?x2xf32>, tensor<?xi32>)
+
+// CHECK-NEXT: ^bb1(%[[BB1_ARG0:.+]]: tensor<4x?x?x2xf32>, %[[BB1_ARG0_D1:.+]]: index, %[[BB1_ARG0_D2:.+]]: index,
+// CHECK-SAME:      %[[BB1_ARG1:.+]]: tensor<?xi32>, %[[BB1_ARG1_D0:.+]]: index)
+^bb1(%bb1_arg0: tensor<4x?x?x2xf32>, %bb1_arg1: tensor<?xi32>):
+  // CHECK-NEXT: %[[TIED_BB1_ARG0:.+]] = flow.tensor.tie_shape %[[BB1_ARG0]] : tensor<4x?x?x2xf32>{%[[BB1_ARG0_D1]], %[[BB1_ARG0_D2]]}
+  // CHECK-NEXT: %[[TIED_BB1_ARG1:.+]] = flow.tensor.tie_shape %[[BB1_ARG1]] : tensor<?xi32>{%[[BB1_ARG1_D0]]}
+
+  // CHECK-NEXT: util.do_not_optimize(%[[TIED_BB1_ARG0]])
+  util.do_not_optimize(%bb1_arg0) : tensor<4x?x?x2xf32>
+  // CHECK-NEXT: util.do_not_optimize(%[[TIED_BB1_ARG1]])
+  util.do_not_optimize(%bb1_arg1) : tensor<?xi32>
+
+  return
+}
+
+// -----
+
+// Tests that selects of dynamically shaped tensors expand to selecting dims.
+
+// CHECK-LABEL: @select
+// CHECK-SAME: (%[[COND:.+]]: i1,
+// CHECK-SAME:  %[[ARG0:.+]]: tensor<4x?x?x2xf32>, %[[ARG0_D1:.+]]: index, %[[ARG0_D2:.+]]: index, %[[ARG1:.+]]: tensor<4x?x?x2xf32>, %[[ARG1_D1:.+]]: index, %[[ARG1_D2:.+]]: index)
+func @select(%cond: i1, %arg0: tensor<4x?x?x2xf32>, %arg1: tensor<4x?x?x2xf32>) {
+  // CHECK-NEXT: %[[TIED_ARG0:.+]] = flow.tensor.tie_shape %[[ARG0]] : tensor<4x?x?x2xf32>{%[[ARG0_D1]], %[[ARG0_D2]]}
+  // CHECK-NEXT: %[[TIED_ARG1:.+]] = flow.tensor.tie_shape %[[ARG1]] : tensor<4x?x?x2xf32>{%[[ARG1_D1]], %[[ARG1_D2]]}
+
+  // CHECK-NEXT: %[[SEL_TENSOR:.+]] = arith.select %[[COND]], %[[TIED_ARG0]], %[[TIED_ARG1]]
+  // CHECK-NEXT: %[[SEL_D1:.+]] = arith.select %[[COND]], %[[ARG0_D1]], %[[ARG1_D1]]
+  // CHECK-NEXT: %[[SEL_D2:.+]] = arith.select %[[COND]], %[[ARG0_D2]], %[[ARG1_D2]]
+  // CHECK-NEXT: %[[SEL_TIED:.+]] = flow.tensor.tie_shape %[[SEL_TENSOR]] : tensor<4x?x?x2xf32>{%[[SEL_D1]], %[[SEL_D2]]}
+  %0 = arith.select %cond, %arg0, %arg1 : tensor<4x?x?x2xf32>
+
+  // CHECK-NEXT: util.do_not_optimize(%[[SEL_TIED]])
+  util.do_not_optimize(%0) : tensor<4x?x?x2xf32>
+
+  return
+}
```

Comment:
fyi there is no newline character at the end of this file (check your editor settings?)

Reply:
thought we had a check for that?